File: composeApp/src/desktopMain/kotlin/com/katsukosail/soundbox/database/dao/PlaylistHasCancionDAO.java
package com.katsukosail.soundbox.database.dao;

import com.katsukosail.soundbox.database.DatabaseConnection;
import com.katsukosail.soundbox.database.model.PlaylistHasCancion;
import java.sql.*;
import java.util.ArrayList;
import java.util.List;


public class PlaylistHasCancionDAO {
    public void insert(PlaylistHasCancion phc, Connection conn) throws SQLException {
        String sql = "INSERT INTO Playlist_has_Cancion (id_playlist, id_cancion) VALUES (?, ?)";
        try (PreparedStatement pstmt = conn.prepareStatement(sql)) {
            pstmt.setInt(1, phc.getIdPlaylist());
            pstmt.setInt(2, phc.getIdCancion());
            pstmt.executeUpdate();
        }
    }
    public List<PlaylistHasCancion> getAll(Connection conn) {
        List<PlaylistHasCancion> lista = new ArrayList<>();
        String sql = "SELECT * FROM Playlist_has_Cancion";
        try (Statement st = conn.createStatement();
             ResultSet rs = st.executeQuery(sql)) {
            while (rs.next()) {
                lista.add(new PlaylistHasCancion(
                        rs.getInt("id_playlist"),
                        rs.getInt("id_cancion")
                ));
            }
        } catch (SQLException e) {
            System.out.println("Error obteniendo relaciones playlist-cancion: " + e.getMessage());
        }
        return lista;
    }

    public PlaylistHasCancion findById(int idPlaylist, int idCancion) {
        String sql = "SELECT * FROM Playlist_has_Cancion WHERE id_playlist = ? AND id_cancion = ?";
        try (Connection conn = DatabaseConnection.connect();
             PreparedStatement stmt = conn.prepareStatement(sql)) {
            stmt.setInt(1, idPlaylist);
            stmt.setInt(2, idCancion);
            try (ResultSet rs = stmt.executeQuery()) {
                if (rs.next()) {
                    return new PlaylistHasCancion(
                            rs.getInt("id_playlist"),
                            rs.getInt("id_cancion")
                    );
                }
            }
        } catch (SQLException e) {
            System.out.println("Error buscando relación playlist-canción: " + e.getMessage());
        }
        return null;
    }

    public void deleteById(int idPlaylist, int idCancion, Connection conn) {
        String sql = "DELETE FROM Playlist_has_Cancion WHERE id_playlist = ? AND id_cancion = ?";
        try (PreparedStatement pstmt = conn.prepareStatement(sql)) {
            pstmt.setInt(1, idPlaylist);
            pstmt.setInt(2, idCancion);
            pstmt.executeUpdate();
        } catch (SQLException e) {
            System.out.println("Error eliminando relación playlist-canción: " + e.getMessage());
        }
    }


}

File: composeApp/src/desktopMain/kotlin/com/katsukosail/soundbox/database/dao/PlaylistHasCancionDAO.kt
package com.katsukosail.soundbox.database.dao

import com.katsukosail.soundbox.database.DatabaseConnection.connect
import com.katsukosail.soundbox.database.model.PlaylistHasCancion
import java.sql.Connection
import java.sql.SQLException

class PlaylistHasCancionDAO {
    @Throws(SQLException::class)
    fun insert(phc: PlaylistHasCancion, conn: Connection) {
        val sql = "INSERT INTO Playlist_has_Cancion (id_playlist, id_cancion) VALUES (?, ?)"
        conn.prepareStatement(sql).use { pstmt ->
            pstmt.setInt(1, phc.idPlaylist)
            pstmt.setInt(2, phc.idCancion)
            pstmt.executeUpdate()
        }
    }

    fun getAll(conn: Connection): MutableList<PlaylistHasCancion?> {
        val lista: MutableList<PlaylistHasCancion?> = ArrayList<PlaylistHasCancion?>()
        val sql = "SELECT * FROM Playlist_has_Cancion"
        try {
            conn.createStatement().use { st ->
                st.executeQuery(sql).use { rs ->
                    while (rs.next()) {
                        lista.add(
                            PlaylistHasCancion(
                                rs.getInt("id_playlist"),
                                rs.getInt("id_cancion")
                            )
                        )
                    }
                }
            }
        } catch (e: SQLException) {
            println("Error obteniendo relaciones playlist-cancion: " + e.message)
        }
        return lista
    }

    fun findById(idPlaylist: Int, idCancion: Int): PlaylistHasCancion? {
        val sql = "SELECT * FROM Playlist_has_Cancion WHERE id_playlist = ? AND id_cancion = ?"
        try {
            connect().use { conn ->
                conn!!.prepareStatement(sql).use { stmt ->
                    stmt.setInt(1, idPlaylist)
                    stmt.setInt(2, idCancion)
                    stmt.executeQuery().use { rs ->
                        if (rs.next()) {
                            return PlaylistHasCancion(
                                rs.getInt("id_playlist"),
                                rs.getInt("id_cancion")
                            )
                        }
                    }
                }
            }
        } catch (e: SQLException) {
            println("Error buscando relación playlist-canción: " + e.message)
        }
        return null
    }

    fun deleteById(idPlaylist: Int, idCancion: Int, conn: Connection) {
        val sql = "DELETE FROM Playlist_has_Cancion WHERE id_playlist = ? AND id_cancion = ?"
        try {
            conn.prepareStatement(sql).use { pstmt ->
                pstmt.setInt(1, idPlaylist)
                pstmt.setInt(2, idCancion)
                pstmt.executeUpdate()
            }
        } catch (e: SQLException) {
            println("Error eliminando relación playlist-canción: " + e.message)
        }
    }
}
